### ISSUE TITLE
TransfoXLLMHeadModel output interpretation

### ISSUE THREAD

**Issue by bergen** [huggingface/pytorch-pretrained-BERT#318]
TransfoXLLMHeadModel gives an output of log probabilities of shape [batch_size, sequence_length, n_tokens]. What do these probabilities represent? For example, what distribution is output at the first sequence position? Is it the conditional distribution given the first word? If so, how can the probability of a complete sentence be computed, including the first word? 

Also, the readme states:
> softmax_output: output of the (adaptive) softmax:
>if target is None: Negative log likelihood of shape [batch_size, sequence_length]

This appears to be incorrect. From current behavior, it should say: if target is **not** None

**Comment by thomwolf** [comment]
Hi,
1/ it's the usual language modeling probabilities: each token probability given the previous tokens
2/ thanks, fixed.
